docs/services/postgresql/split-partitioned-database.md

```diff
@@ -0,0 +1,275 @@
+# Splitting a shard in postgresql
+
+This document describes the process required to split a partitioned database
+from one PostgreSQL instance into itself and another. This migration will
+require downtime.
+
+## Assumptions
+
+For the purposes of this document we'll assume that we have three database machines, *pg1*
+, *pg2* and *pg3*. *pg1* has one database and *pg2* and *pg3* has none:
+
+*pg1* databases:
+
+* *commcarehq_p1* (with django alias *p1*)
+
+*pg2* and *pg3* is a newly deployed server in the *[postgresql]* group and we want to
+create a new  *commcarehq_p2* on *pg2*  and *commcarehq_p3* on *pg3* with half the data from *commcarehq_p1* on each.
+
+*pg1* is currently the only database containing sharded data.
+Half of the data should be moved to a new *pg2* and *pg3* servers
+
+Current database configuration:
+
+```python
+PARTITION_DATABASE_CONFIG = {
+    'shards': {
+        'p1': [0, 3],
+    },
+    'groups': {
+        'main': ['default'],
+        'proxy': ['proxy'],
+        'form_processing': ['p1'],
+    },
+}
+
+DATABASES = {
+    'proxy': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq_proxy',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'pg1',
+        'PORT': '5432',
+        'TEST': {
+            'SERIALIZE': False,
+        },
+    },
+    'p1': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq_p1',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'pg1',
+        'PORT': '5432',
+        'TEST': {
+            'SERIALIZE': False,
+        },
+    },
+}
+```
+
+At the end of this process shards 0 & 1 should be on *pg2* and shards 2 & 3 will be on *pg3*.
+
+## Important Notes
+
+By default pglogical does not replicate any DDL commands.
+This means that any CommCareHQ migrations may not be applied to the target databases while logical replication is active.
+It is recommended to not deploy any changes during the time when splitting the database.
```

Comment:
```suggestion
It is recommended to not deploy any database schema changes during the time when splitting the database.
```

Reply:
I intentionally said any changes because its easier to convey that idea and is guaranteed to be safe. Database schema changes are safe if they aren't on the same database that's being replicated. What do you think about keeping the general "no changes allowed" tone versus the harder to define, more technically correct languages?

Reply:
No, fine to leave as is.